ProcessLib/UncoupledProcessesTimeLoop.cpp

```diff
@@ -343,6 +360,11 @@ std::unique_ptr<UncoupledProcessesTimeLoop> createUncoupledProcessesTimeLoop(
         //! \ogs_file_param{prj__time_loop__processes}
         config.getConfigSubtree("processes"), processes, nonlinear_solvers);
 
+    if (coupling_config)
+    {
+        assert(global_coupling_conv_criteria.size() == per_process_data.size());
```

Comment:
Thanks for adding the check. Actually I'd prefer using OGS_FATAL, because this is a cheap check which is performed only once and the outcome depends on user input. Therefore the check should also be performed in release mode.

Reply:
Changed to use OGS_FATAL.